jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessor.java

```diff
@@ -49,4 +67,97 @@ public static JarType determineJarType(Path jarPath) throws IOException {
     }
     return JarType.STANDARD;
   }
+
+  /**
+   * Explode jar and create three layers for classes, resources and dependencies on container.
+   *
+   * @param jarPath path to jar file.
+   * @param tempDirPath path to temporary jib local directory to use.
+   * @return list of {@link FileEntriesLayer}.
+   * @throws IOException if I/O error occurs when opening the jar file or if the directory for the
+   *     temporary directory path provided, doesn't exist.
+   */
+  public static List<FileEntriesLayer> explodeStandardJar(Path jarPath, @Nullable Path tempDirPath)
+      throws IOException {
+    Path localExplodedJarRoot;
+    if (tempDirPath == null) {
+      try (TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider()) {
+        localExplodedJarRoot = tempDirectoryProvider.newDirectory();
+      }
+    } else {
+      localExplodedJarRoot = tempDirPath;
+    }
+    ZipUtil.unzip(jarPath, localExplodedJarRoot);
+
+    List<FileEntriesLayer> layers = new ArrayList<>();
+    Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
+    Predicate<Path> isResourceFile = isClassFile.negate();
+
+    // Determine class and resource files in the directory containing jar contents and create
+    // FileEntriesLayer for each type of layer (class or resource), while maintaining the
+    // file's original project structure.
+    FileEntriesLayer classesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                localExplodedJarRoot,
+                isClassFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("Classes")
```

Comment:
Minor point, but Jib build plugins use non-capitalized name "classes" (as defined in `JavaContainerBuilder.LayerType`).

Reply:
I added an enum for layer types, similar to what's being done in `JavaContainerBuilder`.